plugin.video.rocketbeans/resources/lib/plugin.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+import routing
+
+from resources.data import config
+from resources.lib.youtube import YoutubeStream
+from resources.lib.twitch import TwitchStream
+from xbmcgui import ListItem
+from xbmcplugin import addDirectoryItem, endOfDirectory, setContent
+from xbmcaddon import Addon
+
+
+plugin = routing.Plugin()
+setContent(plugin.handle, 'videos')
+
+livestream_thumbnail = None
+livestream_url = None
+
+__addon__ = Addon()
+
+def getString(string_id):
+    return __addon__.getLocalizedString(string_id).encode('utf-8', 'ignore')
+
+def get_livestream():
+    if __addon__.getSetting("stream") == 0:
+        t = TwitchStream(config.TWITCH_USER_LOGIN)
+        livestream_url, title, livestream_thumbnail = t.url, t.title, t.thumbnail
+    else:
+        livestream_url, title, livestream_thumbnail = YoutubeStream().get_live_video_info_from_channel_id(config.CHANNEL_ID)
+    return livestream_url, livestream_thumbnail, title
+
+@plugin.route('/')
+def index():
+    livestream_url, livestream_thumbnail, title = get_livestream()
+    li = createListItem(
+        ('Live | %s' % title),
```

Comment:
This is Python 3.7, you can do f-strings :)

Reply:
I just wanted to have it work. Did not really look at all the new features that are available. I'll try to keep it in mind. But I'm probably doing "low effort ports" for now.